src/types/index.ts

```diff
@@ -376,6 +376,29 @@ export interface TextFileOnChangeEvent {
   latestContent: string;
 }
 
+export interface WatchDirListeners {
+  onChange: WatchDirOnChangeListener;
+  onMoveOrDelete?: WatchDirOnMoveOrDeleteListener;
+  onError: WatchDirOnErrorListener;
+}
+
+export enum FsErrors {
+  NotFound = "NOT_FOUND",
+  AlreadyExists = "ALREADY_EXIST",
+  NotDirectory = "NOT_DIRECTORY",
+  IsDirectory = "IS_DIRECTORY",
+  InvalidPath = "INVALID_PATH",
+  DiskQuotaExceeded = "DISK_QUOTA_EXCEEDED",
+}
+
+export type WatchDirOnErrorListener = (
+  err: Error & { code?: FsErrors.NotFound | FsErrors.NotDirectory },
+  extraInfo?: Record<string, any>
+) => void;
+export type WatchDirOnChangeListener = (children: Array<FsNode>) => void;
```

Comment:
I think this is wrong. `watchDir` emits `FsChild` which is `{ filename: string, type: FileType }`, the difference is that it doesn't emit full `path`s, only the trailing component

The interface in repl-it-web is confusing because it's `filename` but it could be a folder lol